### DISCUSSION TITLE
Home Assistant AddOn - Keine Kommunikation mit Geräten

### DISCUSSION THREAD

**Discussion by TheRealSimon42** [OpenCCU/OpenCCU#1833] [🇩🇪 Fragen&Antworten] [answered]
Hallo,
ich habe ein Problem mit dem Home Assistant Add-on und wollte fragen, ob jemand schon mal etwas Ähnliches erlebt hat oder einen Tipp für mich hat.

Situation: 
- Komme von der **CCU2**. Funktioniert noch, ist aber aktuell aus. 
- Druckfrischer Homeassistant (gestern installiertes OS) auf Raspberry Pi 4.
- RaspberryMatic Add-on installiert, der Anleitung gefolgt (ist ja nicht erforderlich aus ein wenig klicken) und **Sicherung der CUU2 eingespielt** (mit Sicherheitsschlüssel, ohne CCU-AddOns, CCU2 war schon aus).

Nach dem Einspielen der Sicherung musste ich (scheinbar) erst mal nach **neuen Geräten suchen** und meinen **HmIP-RFUSB** wieder als Gerät **hinzufügen**. Glaube, der Schritt war so in der Anleitung nicht beschrieben.
<img width="1432" alt="image" src="https://user-images.githubusercontent.com/62359912/167722760-5e8df277-1f5f-4240-b1a7-6a8ea9aefd1e.png">

Wie auch immer, nachdem ich das getan hatte, war (weiterhin) keine Kommunikation mit den wiederhergestellten Geräten möglich & (neu) der Carrier Sense steht auf 100 %.
![2022-05-10_23-16-59 (1)](https://user-images.githubusercontent.com/62359912/167723409-2a6c84fa-b4e3-454e-b1e9-6d0af68d10cd.gif)

Das Log vom AddOn in Home Assistant sieht aber sauber aus:

```
Mounting /data as /usr/local (Home Assistant Add-On): OK
Starting watchdog...
Identifying host system: oci, OK
Initializing RTC Clock: no hardware found
Running sysctl: OK
Checking for Factory Reset: not required
Checking for Backup Restore: not required
Initializing System: OK
Starting logging: OK
Init onboard LEDs: init, OK
Starting irqbalance: OK
Starting iptables: OK
Starting network: eth0: link up, fixed, firewall, inet up, 172.30.33.0, OK
Identifying Homematic RF-Hardware: ....HmRF: HMIP-RFUSB/eQ-3 HmIP-RFUSB@usb-0000:01:00.0-1.3, HmIP: HMIP-RFUSB/eQ-3 HmIP-RFUSB@usb-0000:01:00.0-1.3, OK
Updating Homematic RF-Hardware: HMIP-RFUSB: 4.4.16, not necessary, OK
Starting hs485dLoader: disabled
Starting xinetd: OK
Starting eq3configd: OK
Starting lighttpd: OK
Starting ser2net: disabled
Starting ssdpd: OK
Starting sshd: OK
Starting ha-proxy: OK
Starting NUT services: disabled
Initializing Third-Party Addons: OK
Starting LGWFirmwareUpdate: ...OK
Setting LAN Gateway keys: OK
Starting hs485d: disabled
Starting multimacd: .OK
Starting rfd: .OK
Starting HMIPServer: .............OK
Starting ReGaHss: .OK
Starting CloudMatic: OK
Starting NeoServer: disabled
Starting Third-Party Addons: OK
Starting crond: OK
Setup onboard LEDs: booted, OK
Finished Boot: 3.63.9.20220430 (raspmatic_oci_arm64)
```

Das Debug-Log vom RaspberryMatic CCU auch relativ, so wie ich das als absoluter RaspberryMatic noob lesen kann:
[https://www.icloud.com/iclouddrive/030Impbsx_AbOZGPBn-ZLWJhw#de838cd8-raspberrymatic-2022-05-10_(9)](https://www.icloud.com/iclouddrive/030Impbsx_AbOZGPBn-ZLWJhw#de838cd8-raspberrymatic-2022-05-10_(9))

Meine Frage oder Ideen dazu wären:

- Kann es sein, dass meine Geräte das Problem sind, weil diese permanent mit dem Stick quatschen wollen, er aber einen auf Olaf Scholz macht und es der Carrier Sense abdreht?
- Wie kann ich herausfinden, ob der Stick defekt ist oder kann man schon am Log absehen, dass er es ist? (Ist ein ganz neuer, heute von ELV geliefert und direkt zusammengelötet)
- Habe ich etwas Offensichtliches übersehen?
- Muss ich noch etwas einstellen, um über den Stick die "Klassischen" Homematic Geräte steuern zu können?
- Hat jemand eine andere Idee?

Vielen Dank im Voraus für eure Zeit & Ideen

**Comment by jens-maus** [comment, marked as answer]
> * Druckfrischer Homeassistant (gestern installiertes OS) auf Raspberry Pi 4.

Das (der Einsatz eines RaspberryPi4) in Kombination zu dem im Screencast zu erkennenden CarrierSense von 100% erklärt alle deine hier genannten Probleme sehr gut.

Der RaspberryPi4 ist allgemein dafür bekannt das dieser massive Funkstörungen in seiner näheren Umgebung verursacht. Genau deshalb muss man (wenn man unbedingt ein RaspberryPi4 einsetzen will/muss) einige Dinge beachten die man auch hier nachlesen kann:

https://homematic-forum.de/forum/viewtopic.php?f=31&t=65293

Zusammengefasst ist es absolut notwendig den HmIP-RFUSB so weit wie möglich vom RaspberryPi4 weg zu bringen (z.B. mit einer 2 Meter langen USB-Verlängerung) und auch zusätzlich sicherzustellen das keine der USB3 Ports (die blauen) verwendet werden.

Im Homematic Forum findet man zu dieser Problematik (RaspberryPi4 und Funkstörungen) auch viele andere nützliche Hinweise und Tipps.

**Reply by TheRealSimon42** [reply]
Danke Jens, 
primär für die phänomenale/unglaubliche Antwortzeit 👍

Ich werde morgen mal schauen und mich bei Bedarf nochmals melden, allerdings sollte der Abstand mehr als ausreichend sein. Meine CCU2 war näher und hatte keinerlei Probleme:
![image](https://user-images.githubusercontent.com/62359912/167729138-4b8079ad-39fa-438d-996a-a3dd1ddb66cf.png)


USB 2.0 Verlängerung (bewusst darauf geachtet) ist angeschlossen am USB 2.0 Port des Raspberry PI.

Wie erwähnt, ich melde mich morgen noch einmal und berichte 😃

**Reply by TheRealSimon42** [reply]
Guten Morgen & hallo noch mal :)

Ich habe nun das Problem etwas weiter behoben und habe keinen Carrier Sense (0 %) mehr. Das Problem war, dass ich die Antenne im Stick nicht fest genug angedrückt habe und diese so herumgerutscht ist, dass das Ende des Kabels unter der Platine lag 👍 
Jetzt passt das schonmal.

Aber trotzdem ist keine Steuerung irgendeines Homematic Gerätes möglich, bei Schalten kommt immer der Orange Kreis und sonst nichts, kein Status ändert sich und keine Geräte funktionieren.
![42EFEFB2-60CC-405C-AEB7-CB6EA5B5F12D](https://user-images.githubusercontent.com/62359912/167791419-b9ec9f58-a261-4710-b205-f4985a5d9b70.GIF)



Hier im Log-File sollten die letzten Einträge von Schaltversuchen der Aktoren sein:
https://www.icloud.com/iclouddrive/092Bw7Vh4OQCgMMShOU9SZOxw#de838cd8-raspberrymatic-2022-05-11

Ich kann das als RaspberryMatic Anfänger, wie zuvor erwähnt, nicht wirklich gut beurteilen, aber das Log sieht ja vorerst so aus als würde der Stick mit den Homematic Geräten prinzipiell schonmal kommunizieren, oder?
Irgendwelche anderen Ideen?

**Reply by jens-maus** [reply]
> Ich habe nun das Problem etwas weiter behoben und habe keinen Carrier Sense (0 %) mehr. Das Problem war, dass ich die Antenne im Stick nicht fest genug angedrückt habe und diese so herumgerutscht ist, dass das Ende des Kabels unter der Platine lag 👍 Jetzt passt das schonmal.
> Aber trotzdem ist keine Steuerung irgendeines Homematic Gerätes möglich, bei Schalten kommt immer der Orange Kreis und sonst nichts, kein Status ändert sich und keine Geräte funktionieren. 

Das der CarrierSense bei dir jetzt nicht mehr dauerhaft 100% ist für sich passt schonmal viel besser und zeigt das dieser Wert seine Berechtigung hat. Trotzdem würde ich dir empfehlen nun einfach mal schrittweise vorzugehen und nicht deine große alte CCU2 Installation komplett und verbatim via Hauruck umzustellen bevor du nicht die prinzipielle Funktionalität deiner neuen Umgebung aus HomeAssistant+RaspberryMatic Add-on ausreichend evaluiert+getestet hast.

D.h. nimm dir einfach mal einen Aktor/Sensor heraus, mach im RaspberryMatic Addon ein Werksreset und dann beginn den mal anzulernen. Geht der dann (auch an der Stelle an der du ihn brauchst) kannst du noch mit 2-3 anderen fortfahren und wenn das alles funktioniert dann weisst du prinzipiell das der neue Stick und die Kombi aus HA+RaspberryMatic Add-on in dieser Konstellation mit RaspberryPi4 wirklich zum Erfolg führt.

> Hier im Log-File sollten die letzten Einträge von Schaltversuchen der Aktoren sein: https://www.icloud.com/iclouddrive/092Bw7Vh4OQCgMMShOU9SZOxw#de838cd8-raspberrymatic-2022-05-11
>
> Ich kann das als RaspberryMatic Anfänger, wie zuvor erwähnt, nicht wirklich gut beurteilen, aber das Log sieht ja vorerst so aus als würde der Stick mit den Homematic Geräten prinzipiell schonmal kommunizieren, oder? Irgendwelche anderen Ideen?

Nun, wenn der Stick mit DutyCycle+CarrierSense auftaucht dann wird er ja prinzipiell richtig erkannt. Ein Klick auf "Hilfe" sollte dich auch zu einer Seite bringen wo bzgl. HomeMatic und homematicIP Funk auch noch mehr technische Infos zu finden sind die auch darauf schließen lassen sollten das der Stick prinzipiell gehen sollte. Wenn er dann allerdings trotzdem die Geräte nicht anfunken kann bzw. dieser nicht darauf reagieren kann das aber noch immer daran liegen das ggf. die Antenne vielleicht doch einen Schlag abgekommen hat oder ähnliches. Und das lässt sich auch nicht aus irgendwelchen Logfiles herauslesen und nur durch schrittweises vorgehen+Testen (wie oben erwähnt) empirisch ermitteln...
